ibis/__init__.py

```diff
@@ -76,23 +72,33 @@ def __getattr__(name: str) -> BaseBackend:
     the `ibis.backends` entrypoints. If successful, the `ibis.sqlite`
     attribute is "cached", so this function is only called the first time.
     """
-    entry_points = list(
-        pkg_resources.iter_entry_points(group='ibis.backends', name=name)
-    )
-    if len(entry_points) == 0:
+    entry_points = [
+        entry_point
+        for entry_point in importlib_metadata.entry_points().get(
+            "ibis.backends",
+            (),
+        )
+        if entry_point.name == name
+    ]
```

Comment:
This seems to be double-counting all the backends for a reason I don't understand. It would be a bit gross but if we list(set( it then it should work

Reply:
I'm using the `importlib.metadata.distribution` API now to avoid picking up multiple names. I think this shows up as both `ibis-framework` _and_ `ibis` when installed with `setuptools`.